### PR TITLE
Allowance bugs and enhancements

### PR DESCRIPTION
**Pull request by akash2017sky**
Fixed the bugs and enhancements:

[Enhancement 90929](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/90929): Font sizes are wrong on the wallet page
[Bug 91016](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91016): Send and Receive transaction history shows inconsistent values.
[Enhancement 90809](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/90809): Consistently aligned "Sats"
[Enhancement 90849](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/90849): Reposition refresh icon
[Enhancement 90850](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/90850): Remove blue square by balance Sats
[Task 90532](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/90532): Build transaction history

Features:
[Feature 89525](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/89525): React App - Allowance Component
[Feature 89981](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/89981): React App - Wallet - Transaction History

### CODE REVIEW

**Comment by BenGWeeks** [comment]
Please test this @ediweeks.

File: tabs/src/Wallet.tsx [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
           availableSats={2000000}
           availableAmountUSD={74.2}
           remainingSats={90000}
           spentSats={20000}
         />
         <WalletTransactionHistory />
         <WalletTransactionLog />
+         */}
+
+        <WalletAllwanaceMainComponent />
```
Comment (BenGWeeks) [inline tabs/src/Wallet.tsx:18]: Typo.

Comment (BenGWeeks) [inline tabs/src/Wallet.tsx:18]: Suggest we pass in the wallet ID as a property into the Component. E.g. `<WalletAllowanceComponent WalletId="1234567890"/>`

File: tabs/src/YourWallet.tsx [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
-      <WalletYourWalletInfoCard />
-      <WalletTransactionHistory />
-      <WalletTransactionLog />
+    // <div className="app-container">
+    //   <WalletYourWalletInfoCard />
+    //   <WalletTransactionHistory />
+    //   <WalletTransactionLog />
+    // </div>
+    <div>
+      <WalletAllwanaceMainComponent />
```
Comment (BenGWeeks) [inline tabs/src/YourWallet.tsx:16]: Typo.

File: tabs/src/components/WalletAllowanceCard.css
Comment (BenGWeeks) [inline tabs/src/components/WalletAllowanceCard.css:1]: Use our `*.module.css` form.

File: tabs/src/components/WalletAllwanaceMainComponent.tsx
Comment (BenGWeeks) [inline tabs/src/components/WalletAllwanaceMainComponent.tsx:1]: Typo

**Review by BenGWeeks — CHANGES_REQUESTED** [review]

**Comment by BenGWeeks** [comment]
Looks good. Assume Send/Receive coming separately for that (if you have any draft code, submit a draft PR and I can help you with it). @ediweeks please raise the following (can be done as a separate PR):

Bug: Unable to filter by Sent/Received
Assigned to: @akash2017sky 

image

image

Enhancement: Alignment issue on wallet transactions
Assigned to: @akash2017sky 

image

Bug: Conversion is wallet balance isn't implemented
Assigned to: @akash2017sky 

image

File: tabs/src/components/WalletYourWalletInfoCard.tsx [hunk 3]
```diff
@@ -10,45 +10,65 @@ import { getUserWallets } from '../services/lnbitsServiceLocal';
...
-
   const [balance, setBalance] = useState<number>();
-  
+
   const fetchAmountReceived = async () => {
     console.log('Fetching your wallet ...');
-    const walletForUser = await getUserWallets(adminKey,'2984e3ac627e4fea9fd6dde9c4df83b5');
+    const walletForUser = await getUserWallets(
+      adminKey,
+      '2984e3ac627e4fea9fd6dde9c4df83b5',
```
[the remark this answers is not in the document]
Comment (BenGWeeks) [reply tabs/src/components/WalletYourWalletInfoCard.tsx:19]: I.e. looks something like like:

```
const currentUser = await getCurrentUser();
const privateWallet = currentUser .privateWallet`
```

Where `getCurrentUser()` is defined in a new `/services/UserService.ts` service (which basically talks to the Microsoft Graph API I assume when SSO implemented). NB `getCurrentUser()` returns a `User` type, whilst `privateWallet` is a `Wallet` type (as defined in `/types/global.d.ts`.

FYI @desousamario85 @NataliaMilgevskaya

File: tabs/src/services/lnbitsServiceLocal.ts
[the remark this answers is not in the document]
Comment (BenGWeeks) [reply tabs/src/services/lnbitsServiceLocal.ts:1]: Discussed this with @desousamario85. Let's remove this from this PR, and create a new PR for proxy removal (which needs additional changes such as deleting `setupProxy.js`).

**Comment by EdiWeeks** [comment]
Done
[Bug 91576](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91576): Unable to filter by Sent/Received
[Enhancement 91577](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91577): Alignment issue on wallet transactions
[Bug 91578](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91578): Conversion is wallet balance isn't implemented

File: tabs/src/YourWallet.tsx
Comment (BenGWeeks) [inline tabs/src/YourWallet.tsx:1]: Just one `Wallet` page please. You can switch between Allowance and Private views just with a variable.

**Review by BenGWeeks — CHANGES_REQUESTED** [review]

**Comment by EdiWeeks** [comment]
@akash2017sky please take a look at these:
[Bug 91647](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91647): Sats are different sizes
[Bug 91648](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91648): Receive and Send buttons are moving when reducing the window size

File: tabs/src/Wallet.tsx (hunk 1 above)
Comment (akash2017sky) [reply tabs/src/Wallet.tsx]: Updated the the class name

File: tabs/src/YourWallet.tsx (hunk 2 above)
Comment (akash2017sky) [reply tabs/src/YourWallet.tsx]: Updated the the file name

File: tabs/src/YourWallet.tsx
Comment (akash2017sky) [reply tabs/src/YourWallet.tsx:1]: Deleted the YourWallet.tsx

File: tabs/src/components/WalletAllowanceCard.tsx [hunk 4]
```diff
@@ -34,16 +37,29 @@ const WalletAllowanceCard: React.FC<AllowanceCardProps> = ({
 
   const fetchAmountReceived = async () => {
     console.log('Fetching your wallet ...');
-    const walletForUser = await getUserWallets(adminKey, '2984e3ac627e4fea9fd6dde9c4df83b5');
 
-    console.log('Wallets: ', walletForUser);
+    const currentUserLNbitDetails = await getUsers(adminKey, {
```
Comment (BenGWeeks) [inline tabs/src/components/WalletAllowanceCard.tsx:41]: I don't think you need to do this. Just call `getUser`, then `currentUserLNbitsDetails.allowanceWallet.balance_msat`

File: tabs/src/components/WalletTransactionHistory.tsx
Comment (BenGWeeks) [inline tabs/src/components/WalletTransactionHistory.tsx:1]: Why do we need `WalletTransactionHistory` as well as `WalletTransactionLog`?

File: tabs/src/components/WalletYourWalletInfoCard.css
Comment (BenGWeeks) [inline tabs/src/components/WalletYourWalletInfoCard.css:1]: This should be a module, `WalletYourWalletInfoCard.module.css` (see how done elsewhere).

File: tabs/src/components/WalletAllowanceCard.css
Comment (BenGWeeks) [inline tabs/src/components/WalletAllowanceCard.css:1]: This should be a module really.

File: tabs/src/components/WalletYourWalletInfoCard.css
Comment (BenGWeeks) [reply tabs/src/components/WalletYourWalletInfoCard.css:1]: Just `WalletInfoCard.module.css`

File: tabs/src/components/WalletYourWalletInfoCard.tsx
Comment (BenGWeeks) [inline tabs/src/components/WalletYourWalletInfoCard.tsx:1]: Just `WalletInfoCard.tsx`

File: tabs/src/services/lnbitsServiceLocal.ts
Comment (BenGWeeks) [inline tabs/src/services/lnbitsServiceLocal.ts:1]: If you're doing this (using `${nodeUrl}`), you should remove `setupProxy.js` and confirm it works as intended.

**Review by BenGWeeks — CHANGES_REQUESTED** [review]

**Comment by BenGWeeks** [comment]
Enhancement: Add 20px margin to the top
image

Enhancement: Add a 40px margin to the bottom
image

Enhancement: Missing empty (for now) components on wallets screens
image
image
(We can leave that if takes significant time for now)

FYI @EdiWeeks

**Comment by BenGWeeks** [comment]
Bug: Sent/Received not filtering
image

File: tabs/src/components/WalletAllowanceMainComponent.tsx
Comment (BenGWeeks) [inline tabs/src/components/WalletAllowanceMainComponent.tsx:1]: I don't think we a `WalletAllowanceMainComponent` and `WalletAllowanceCard` component. Just one `WalletAllowanceComponent`.

**Review by BenGWeeks — CHANGES_REQUESTED** [review]

**Comment by BenGWeeks** [comment]
Bug: Font is incorrect size (changes size between Your wallet and Allowance)
image

File: tabs/src/components/WalletAllowanceCard.tsx (hunk 4 above)
Comment (akash2017sky) [reply tabs/src/components/WalletAllowanceCard.tsx:41]: Interesting.. yes.. updated this for Private and Allowance wallet too.

**Comment by EdiWeeks** [comment]
[Enhancement 91778](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91778): Add 20px margin to the top
[Enhancement 91779](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91779): Add a 40px margin to the bottom
[Enhancement 91780](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91780): Missing empty (for now) components on wallets screens

**Comment by EdiWeeks** [comment]
We have these bugs already raised previously @BenGWeeks 
[Bug 91017](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91017): Tabs Send and Receive are not clickable - filter not working
[Bug 91647](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91647): Sats are different sizes

**Comment by BenGWeeks** [comment]
Enhancement: Merge `WalletTransactionLog` and `WalletTransactionHistory`
image

Task: `getAllowance` in `lnbitsService.ts` is not implemented
Tag: Blocked
Assigned: @akash2017sky 

image

Task: `getCurrentUser` in `userService.ts` is not implemented
Assigned: @akash2017sky 

image

FYI @EdiWeeks

**Pull request merged**